src/couch_changes.erl

```diff
@@ -336,6 +343,16 @@ get_doc_ids(_) ->
     throw({bad_request, no_doc_ids_provided}).
 
 
+get_selector({json_req, {Props}}) ->
+    check_selector(couch_util:get_value(<<"selector">>, Props));
+get_selector(#httpd{method='POST'}=Req) ->
+    couch_httpd:validate_ctype(Req, "application/json"),
+    {Props} = couch_httpd:json_body_obj(Req),
+    check_selector(couch_util:get_value(<<"selector">>, Props));
```

Comment:
Last line: `get_selector`, not `check_selector`. Also, point about function_clause is false: I miss the moment that json_body_obj already handles all the routines to ensure that we got JSON object, nothing else.